Stats Revamp v2: Add action types

ravishanker opened the pull request:
Adds types for action cards

Fixes [#16581](https://github.com/wordpress-mobile/WordPress-Android/issues/16581)
This can be tested through [#16608](https://github.com/wordpress-mobile/WordPress-Android/pull/16608)

ravishanker committed:
Add action types

ravishanker committed:
Merge branch 'trunk' into Stats-Revamp-v2-Add-action-card-types

ravishanker committed:
Update StatsStore.kt

ravishanker committed:
Merge branch 'trunk' into Stats-Revamp-v2-Add-action-card-types

irfano commented on fluxc/src/main/java/org/wordpress/android/fluxc/store/StatsStore.kt:
```diff
@@ -75,10 +82,73 @@ class StatsStore
             }
 
+    fun hideActionCard(type: ActionType) =
+            coroutineEngine.run(AppLog.T.STATS, this, "hideInsightsReminderActionCard") {
```
loggedMessage is always `hideInsightsReminderActionCard`. We can add `ActionType` parameter to loggedMessage. Also same in `isActionCardShowing`

irfano reviewed (approved):
What `hideActionCard()` and `isActionCardShowing()` do?  
Since this is a persistence library, not related to UI, maybe you could prefer function names like `setActionCharShown` or `isActionCardShown`, up to you, just np.

Feel free to merge after fixing the loggedMessage problem.

ravishanker committed:
update log message with type

ravishanker committed:
Refactor logic

Updated logic as per clarification
- Action cards appear only once per site
- they are removed once user takes either dismiss or card action

ravishanker commented:
> What `hideActionCard()` and `isActionCardShowing()` do? Since this is a persistence library, not related to UI, maybe you could prefer function names like `setActionCharShown` or `isActionCardShown`, up to you, just np.
> 
> Feel free to merge after fixing the loggedMessage problem.

Good point.  However, I had to remove that completely after reworking logic based on clarification on channel
